app/src/main/java/com/mho/training/sources/ReviewDataSource.kt

```diff
@@ -15,24 +15,24 @@ import java.io.IOException
 
 class ReviewDataSource(private val resources: Resources) : RemoteReviewDataSource {
 
-    override suspend fun getReviewList(movieId: Int): Result<List<Review>> = withContext(Dispatchers.IO) {
+    override suspend fun getReviewList(movieId: Int): DataResult<List<Review>> = withContext(Dispatchers.IO) {
         safeApiCall(
             call = { requestReviewList(movieId) },
             errorMessage = resources.getString(R.string.error_unable_to_fetch_reviews)
         )
     }
 
-    private suspend fun requestReviewList(movieId: Int): Result<List<Review>> {
+    private suspend fun requestReviewList(movieId: Int): DataResult<List<Review>> {
         val response = ReviewRequest.service
             .getReviewListByMovieAsync(movieId, BuildConfig.MOVIE_DB_API_KEY)
 
         if(response.isSuccessful){
```

Comment:
You could avoid the nested if by 
```suggestion

        if(!response.isSuccessful) {
            return Result.Error(IOException(resources.getString(R.string.error_during_fetching_reviews)))
        }
        val results = response.body()?.results
        if (results.isNullOrEmpty()) {
            return Result.Error(IOException(resources.getString(R.string.error_response_is_null)))
        }
        return Result.Success(results.map { it.toDomainReview() })
```

[stop using if](https://medium.com/better-programming/why-you-need-to-stop-using-else-statements-5b1fd09dea9e)

Reply:
I'll analyze how to handle different types of errors, for now, it's ok.